### DISCUSSION TITLE
Multi-day Events

### DISCUSSION THREAD

**Discussion by stelladraco27** [davish/obsidian-full-calendar#508] [Feature Requests]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/davish/obsidian-full-calendar/discussions/categories/feature-requests) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/davish/obsidian-full-calendar/issues) didn't find any similar feature requests.

### 🔖 Feature description

As it is supported in the FullCalendar library, the ability to have an event span multiple days should be supported. 

### ✔️ Solution

Support for the event object 'end' property allows for multi day events. This is useful if an event spans multiple days, such as a vacation, exam week, etc. and you don't want to add a bunch of separate events to span the period. 

### ❓ Alternatives

I have considered multiple events spanning the time period, but this is cumbersome and not easy to manage or visualize.

### 📝 Additional Context

![image](https://github.com/davish/obsidian-full-calendar/assets/24236473/990c6c0c-7b7b-451a-b170-4402c2d2b2e1)
This screenshot is from the FullCalendar website demo page showing that this feature is supported. The particular demo being shown is under "Change The Theme".

**Comment by ghost** [comment]
Isn't this possible? 
![Screenshot from 2023-11-01 15-11-06](https://github.com/davish/obsidian-full-calendar/assets/101370810/da406263-87b4-478f-81b8-da14a2c6d1f2)
![Screenshot from 2023-11-01 15-11-34](https://github.com/davish/obsidian-full-calendar/assets/101370810/80e2b736-4ede-45bf-a8a9-4e21d3037ec7)

**Reply by stelladraco27** [reply]
I will have to check, maybe they put out an update that added functionality since I posted this.

**Comment by Gander7** [comment]
I seem to have found a workaround for this with Full Note Calendars.
I created an allDay note for a day and then added an `endDate` to the frontmatter. This was the result.
It seems like you have to put the endDate, the day AFTER the actual end date.

```md
---
title: Title
allDay: true
date: 2024-07-21
endDate: 2024-07-23
completed: null
---
```

![image](https://github.com/obsidian-community/obsidian-full-calendar/assets/4229134/c9815821-8695-4020-bc67-5d23503b57be)

There is no support in the UI for this but a recommendation, if this is ever officially supported for single events a `date` frontmatter and for multi-day events a `startDate` and `endDate` frontmatter.

EDIT: This doesn't work all the time. I've reverted to using recurring allday events to denote multi-day events.
